sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeFormatterHelper.scala

```diff
@@ -57,6 +60,27 @@ trait DateTimeFormatterHelper {
     }
     formatter
   }
+
+  // When legacy time parser policy set to EXCEPTION, check whether we will get different results
+  // between legacy format and new format. For legacy parser, DateTimeParseException will not be
+  // thrown. On the contrary, if the legacy policy set to CORRECTED, DateTimeParseException will
```

Comment:
`For legacy parser, DateTimeParseException will not be thrown`
I think it should be
`If new parser fails but legacy parser works, throw a SparkUpgradeException.`

Reply:
`On the contrary, if the legacy policy set to CORRECTED ...`

`If the legacy policy set to CORRECTED, do nothing and let the exception propagate.`